system/scripts/zone/core/calc_monster.cs

```diff
@@ -334,4 +334,140 @@ public float SCR_Get_MON_SDR(Mob monster)
+		
+		return (int)Math.Max(0, value);
+	}
+	
+	/// <summary>
+	/// Returns the monster's Critical Dodge Rate (Critical resistance).
+	/// </summary>
+	/// <param name="monster"></param>
+	/// <returns></returns>
+	[ScriptableFunction]
+	public float SCR_Get_MON_CRTDR(Mob monster)
+	{
+		if (monster.Properties.Overrides.TryGetFloat(PropertyName.CRTDR, out var fixValue))
+			return fixValue;
+		
+		var value = (float)monster.Data.CritDodgeRate;
+		
+		var byBuffs = 0f;
+		byBuffs += monster.Properties.GetFloat(PropertyName.CRTDR_BM, 0);
+		
+		var byRateBuffs = 0f;
+		byRateBuffs += monster.Properties.GetFloat(PropertyName.CRTDR_RATE_BM);
+		byRateBuffs = value * byRateBuffs;
+		
+		value = value + byBuffs + byRateBuffs;
+		
+		var decRatio = monster.Properties.GetFloat(PropertyName.CRTDR_RATE_MUL_BM, 1);
+		
+		if (decRatio < 0.5f)
+			decRatio = 0.5f;
+		
+		value *= decRatio;
+		
+		return (int)Math.Max(0, value);
+	}
+
+	/// <summary>
+	/// Returns the monster's Critical Attack.
+	/// </summary>
+	/// <param name="monster"></param>
+	/// <returns></returns>
+	[ScriptableFunction]
+	public float SCR_Get_MON_CRTATK(Mob monster)
+	{
+		if (monster.Properties.Overrides.TryGetFloat(PropertyName.CRTATK, out var fixValue))
+			return fixValue;
+
+		var value = (float)monster.Data.CritAttack;
+		
+		var byBuffs = 0f;
+		byBuffs += monster.Properties.GetFloat(PropertyName.CRTATK_BM, 0);
+		
+		var byRateBuffs = 0f;
+		//byRateBuffs += monster.Properties.GetFloat(PropertyName.CRTATK_RATE_BM);
+		byRateBuffs = value * byRateBuffs;
+		
+		value = value + byBuffs + byRateBuffs;
+		
+		return (int)Math.Max(0, value);
+	}
+
+	/// <summary>
+	/// Returns the monster's block rate.
+	/// </summary>
+	/// <param name="monster"></param>
+	/// <returns></returns>
+	[ScriptableFunction]
+	public float SCR_Get_MON_BLK(Mob monster)
+	{
+		if (monster.Properties.Overrides.TryGetFloat(PropertyName.BLK, out var fixValue))
+			return fixValue;
+		
+		var properties = monster.Properties;
+		
+		var baseValue = (float)monster.Data.BlockRate;
+		var value = baseValue;
+		
+		var byBuffs = properties.GetFloat(PropertyName.BLK_BM);
+		
+		var byRateBuffs = 0f;
+		byRateBuffs += properties.GetFloat(PropertyName.BLK_RATE_BM);
+		byRateBuffs = (value * byRateBuffs);
+		
+		value += byBuffs + byRateBuffs;
+		
+		return value;
+	}
+	
+	/// <summary>
+	/// Returns the monster's block break (penetration).
+	/// </summary>
+	/// <param name="monster"></param>
+	/// <returns></returns>
+	[ScriptableFunction]
+	public float SCR_Get_MON_BLK_BREAK(Mob monster)
+	{
+		if (monster.Properties.Overrides.TryGetFloat(PropertyName.BLK_BREAK, out var fixValue))
+			return fixValue;
+		
+		var properties = monster.Properties;
+		
+		var baseValue = (float)monster.Data.BlockBreakRate;
+		var value = baseValue;
+		
+		var byBuffs = properties.GetFloat(PropertyName.BLK_BREAK_BM);
+		
+		var byRateBuffs = 0f;
+		byRateBuffs += properties.GetFloat(PropertyName.BLK_BREAK_RATE_BM);
```

Comment:
And this one doesn't exist for monsters either. I checked some older data as well, and it doesn't seem like these properties ever existed for monsters. They didn't even exist for players in older versions. I wonder if we should leave them, in case they're ever added, or if we should rather remove them to avoid confusion.

Reply:
Maybe I'll leave them commented for now so we remember they exist but they're not actually implemented?
Or should I just remove them entirely?
What would be the best solution in your opinion?

Reply:
I'm one the fence. They don't exist for monsters currently, so including them in any way could lead to confusion. Either because people don't see them anywhere, or, if they're commented, because one might think they just weren't implemented by us yet. I believe it would be best to remove them until they're actually added for monsters, if ever.